test/regexmanager.cpp

```diff
@@ -106,3 +106,33 @@ TEST_CASE("`highlight all` adds rules for all locations", "[RegexManager]")
 		}
 	}
 }
+
+TEST_CASE("RegexManager does not hang on regexes that can match empty strings", "[RegexManager]")
+{
+	RegexManager rxman;
+	std::string input = "The quick brown fox jumps over the lazy dog";
+	std::string compare = input;
+
+	rxman.handle_action("highlight", {"feedlist", "w*", "blue", "red"});
+	rxman.quote_and_highlight(input, "feedlist");
+	REQUIRE(input == "The quick bro<0>w</>n fox jumps over the lazy dog");
+}
+
+TEST_CASE("RegexManager does not hang on regexes that match empty strings", "[RegexManager]")
+{
+	RegexManager rxman;
+	std::string input = "The quick brown fox jumps over the lazy dog";
+	std::string compare = input;
+
+	rxman.handle_action("highlight", {"feedlist", "$", "blue", "red"});
+	rxman.quote_and_highlight(input, "feedlist");
+	REQUIRE(input == compare);
+
+	rxman.handle_action("highlight", {"feedlist", "^", "blue", "red"});
+	rxman.quote_and_highlight(input, "feedlist");
+	REQUIRE(input == compare);
+
+	rxman.handle_action("highlight", {"feedlist", "^$", "blue", "red"});
```

Comment:
At this point, `rxman` contains three regexes, so if something breaks, it won't be obvious which of the regexes caused the problem. Please put each block into its own `SECTION`, so each test run is using just one regex ([more on sections in the Catch framework's docs](https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#test-cases-and-sections)).